include/unifex/cleanup_adapt_stream.hpp

```diff
@@ -0,0 +1,52 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <unifex/stream_concepts.hpp>
+
+#include <functional>
+#include <type_traits>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+namespace _cleanup_adapt {
+template <typename Stream, typename AdaptFunc>
+struct _stream {
+  struct type;
+};
+template <typename Stream, typename AdaptFunc>
+using stream =
+    typename _stream<remove_cvref_t<Stream>, std::decay_t<AdaptFunc>>::type;
+
+template <typename Stream, typename AdaptFunc>
+struct _stream<Stream, AdaptFunc>::type {
+  UNIFEX_NO_UNIQUE_ADDRESS Stream innerStream_;
+  UNIFEX_NO_UNIQUE_ADDRESS AdaptFunc adapter_;
+
+  friend auto tag_invoke(tag_t<next>, type& s) -> next_sender_t<Stream> {
```

Comment:
This should be conditionally `noexcept`.
```suggestion
  friend auto tag_invoke(tag_t<next>, type& s) noexcept(noexcept(next(s.innerStream_))) -> next_sender_t<Stream> {
```